### PR TITLE
Bump travis to py37

### PR DESCRIPTION
**Pull request by bjlittle**
Switch to using [minimal](https://docs.travis-ci.com/user/languages/minimal-and-generic/) travis containers, optimised for speed and less disk space.

Minior tidy of `.travis.yml`, and bumped coverage to include Python37.

### CODE REVIEW

**Comment by jswhit** [comment]
Looks fine to me.  @ocefpaf - OK with you?

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,54 @@
...
+    conda create --yes -n ${ENV_NAME} python=${PYTHON_VERSION} --file requirements.txt --file requirements-dev.txt;
+    source activate ${ENV_NAME};
+    if [[ "${BUILD_DOCS}" == "true" ]]; then
       conda install -qy sphinx;
     fi;
 
-# Test source distribution.
+  # Output environment details.
+  - >
+    conda list -n ${ENV_NAME};
```
Comment (bjlittle) [inline .travis.yml]: The `conda list -n ${ENV_NAME}` is for humans, as it keeps the packages is alphabetical order.

File: .travis.yml [hunk 2]
```diff
@@ -1,37 +1,54 @@
...
+    conda list -n ${ENV_NAME} --explicit;
```
Comment (bjlittle) [inline .travis.yml]: The `conda list -n ${ENV_NAME} --explicit` is for developers to allow them to easily recreate the travis-ci conda environment locally.

File: .travis.yml [hunk 3]
```diff
@@ -54,5 +71,4 @@ deploy:
     skip_cleanup: true
     on:
       branch: master
-      python: 3.6
-      condition: '$BUILD_DOCS == "true"'
+      condition: '${PYTHON_VERSION} == 3.7 AND ${BUILD_DOCS} == "true"'
```
Comment (bjlittle) [inline .travis.yml]: I think this line is right... haven't seen it run, as it'll need a merge onto master, so I'm assuming it's okay...

**Comment by ocefpaf** [comment]
> Looks fine to me. @ocefpaf - OK with you?

LGMT. Very close to what I'm using for Python 3.7 lately. :+1:

**Comment by jswhit** [comment]
OK, thanks @bjlittle  - merging now.

**Comment by bjlittle** [comment]
Lovely! Thanks @jswhit

**Pull request merged**